reframe/frontend/statistics.py

```diff
@@ -197,6 +199,42 @@ def json(self, force=False):
 
         return self._run_data
 
+    def junit(self, json_report, force=False):
+        # https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
+        xml_testsuites = ET.Element('testsuites')
+        xml_testsuite = ET.SubElement(
+            xml_testsuites, 'testsuite',
+            attrib={
+                'name': 'rfm',
+                'errors': '0',
+                'failures': str(json_report['session_info']['num_failures']),
+                'tests': str(json_report['session_info']['num_cases']),
+                'time': str(json_report['session_info']['time_elapsed']),
+                'hostname': json_report['session_info']['hostname'],
+            }
+        )
+
+        for testid in range(len(json_report['runs'][0]['testcases'])):
```

Comment:
Here you are just taking the first run. How are we going to deal with the retries? Perhaps, we can completely ignore them since their future is not sure, but we should somehow mention it or issue a warning if the `--report-junit` option is passed along with `--max-retries`.

Reply:
For this one we should add a note in the command line option description for the report.